fsi/fsi.go

```diff
@@ -133,6 +133,9 @@ func (fsi *FSI) UpdateLink(dirPath, refStr string) (string, error) {
 }
 
 // Unlink breaks the connection between a directory and a dataset
+// TODO (b5) - should this even need the refStr, since we consider the .qri-ref
+// file the canonical source of truth, shouldn't this look for that file,
+// and use the reference stored within to talk to the repo?
```

Comment:
What if .qri-ref is deleted, corrupted, or otherwise made unreadable (permissions, network filesystem issues), but the ref link still existed in the fbs? We should still handle that case, ignoring the .qri-ref file's error but still updating the repo.

Reply:
Well, that's a great set of reasons not to rely solely on `.qri-ref` 😄